File: JS/problem017.js



var numberLetters = function() {
	
	var onesWords = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];

	var teensWords = ['', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen'];
	
	var tensWords = ['', 'ten', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety'];
	
	for (var h = 0; h < 10; h++) {
		
		var htext = h == 0 ? '': onesWords[h] + 'hundred'
		if (h != 0) console.log(htext);
		
		for (var t = 0; t < 10; t++) {
			
			var ttext = (t == 0 ? '' : tensWords[t]);
			if (t != 0) console.log(htext +'and' + ttext);
			
			for (var o = 1; o < 10; o++) {
				var otext = onesWords[o];
				console.log(t == 1 ? htext + otext);
			}
			
		}
		
	}
	
	
}

numberLetters();

File: python/problem017.py
# If all the numbers from 1 to 1000 (one thousand) inclusive were written out in words, how many letters would be used?

tens = ["TWENTY", "THIRTY", "FORTY", "FIFTY", "SIXTY", "SEVENTY", "EIGHTY", "NINETY"]
hundreds = ["ONE", "TWO", "THREE", "FOUR", "FIVE", "SIX", "SEVEN", "EIGHT", "NINE" ]
numbers = []

numbers.append("ONE")
numbers.append("TWO")
numbers.append("THREE")
numbers.append("FOUR")
numbers.append("FIVE")
numbers.append("SIX")
numbers.append("SEVEN")
numbers.append("EIGHT")
numbers.append("NINE")
numbers.append("TEN")
numbers.append("ELEVEN")
numbers.append("TWELVE")
numbers.append("THIRTEEN")
numbers.append("FOURTEEN")
numbers.append("FIFTEEN")
numbers.append("SIXTEEN")
numbers.append("SEVENTEEN")
numbers.append("EIGHTEEN")
numbers.append("NINETEEN")
    
for num in range(len(tens)):
  numbers.append(tens[num])
  for j in range(9):
    numbers.append(tens[num] + numbers[j])
            
for num in range(9):
  numbers.append(hundreds[num] + "HUNDRED")
  for k in range(19):
    numbers.append(hundreds[num] + "HUNDREDAND" + numbers[k])
  for j in range(8):
    numbers.append(hundreds[num] + "HUNDREDAND" + tens[j])
    for k in range(9):
      numbers.append(hundreds[num] + "HUNDREDAND" + tens[j] + numbers[k])

numbers.append("ONETHOUSAND")

print(sum(len(num) for num in numbers))
